order/views.py

```diff
@@ -1 +1,280 @@
+            return JsonResponse({"message": "DOES_NOT_EXIST"}, status=400)
+
+        except MultipleObjectsReturned:
+            return JsonResponse({"message": "MULTIPLE_OBJECTS_RETURNED"}, status=400)
+
+class CartDetailView(View):
+    @login_decorator
+    def delete(self, request, product_stock_id):
+        try:
+            if not ProductStock.objects.filter(id=product_stock_id):
+                return JsonResponse({"message": "DOES_NOT_EXIST"}, status=404)
+
+            user          = request.user
+            order         = Order.objects.get(user=user, order_status=OrderStatus.objects.get(name='주문 전'))
+            order_product = order.product_stock.get(id=product_stock_id)
+
+            order_product.delete()
+
+            # 장바구니에 상품이 없을 경우 order 정보 삭제
+            if not OrderProductStock.objects.filter(order=order).exists():
+                order.delete()
+
+            return JsonResponse({"message": "SUCCESS"}, status=200) 
+
+        except Order.DoesNotExist:
+            return JsonResponse({"message": "DOES_NOT_EXIST"}, status=400)
+
+        except MultipleObjectsReturned:
+            return JsonResponse({"message": "MULTIPLE_OBJECTS_RETURNED"}, status=400)
+
+        except OrderStatus.DoesNotExist:
+            return JsonResponse({"message": "DOES_NOT_EXIST"}, status=400)
+
+        except OrderProductStock.DoesNotExist:
+            return JsonResponse({"message": "DOES_NOT_EXIST"}, status=400)
+
+class CheckOutView(View):
+    @login_decorator
+    def get(self, request, order_number):
+        user = request.user
+
+        if not Order.objects.filter(order_number=order_number).exists():
+            return JsonResponse({"message": "DOES_NOT_EXIST"}, status=404)
+
+        order_info     = Order.objects.get(order_number=order_number)
+        order_products = order_info.orderproductstock_set.all()
+
+        cart_product_list = [{
+            "category"        : order_product.product_stock.product.category.menu.name,
+            "productId"       : order_product.product_stock.product.id,
+            "productName"     : order_product.product_stock.product.name,
+            "productSubName"  : order_product.product_stock.product.sub_name,
+            "productStockId"  : order_product.product_stock.id,
+            "productSize"     : order_product.product_stock.size,
+            "productPrice"    : order_product.product_stock.price,
+            "productIsSoldOut": bool(order_product.product_stock.stock - order_product.quantity <= 0),
+            "productImageUrl" : order_product.product_stock.product.image_set.get(is_main=True).image_url,
+            "productQuantity" : order_product.quantity
+            } for order_product in order_products
+        ]
+        if 
+         
+        address_info = user.address_set.filter(is_main=True)
+        address      = address_info[0].address if address_info else ""
+        zip_code     = address_info[0].zip_code if address_info else ""
```

Comment:
```suggestion
        address      = address_info.first().address if address_info else ""
        zip_code     = address_info.first().zip_code if address_info else ""
```
queryset에 [0]처럼 인덱스를 직접적으로 사용하는 것은 지양하는 것이 좋습니다.

Reply:
넵 수정했습니다